packages/camera/camera_platform_interface/lib/src/types/media_settings.dart

```diff
@@ -0,0 +1,60 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// ignore_for_file: avoid_equals_and_hash_code_on_mutable_classes
+
+import 'resolution_preset.dart';
+
+/// Recording media settings.
+///
+/// Used in [CameraPlatform.createCameraWithSettings].
+/// Allows to tune recorded video parameters, such as resolution, frame rate, bitrate.
+class MediaSettings {
+  /// Creates a [MediaSettings].
+  const MediaSettings({
+    this.resolutionPreset,
+    this.fps,
+    this.videoBitrate,
+    this.audioBitrate,
+    this.enableAudio = false,
+  });
+
+  /// [ResolutionPreset] affect the quality of video recording and image capture.
+  final ResolutionPreset? resolutionPreset;
+
+  /// Rate at which frames should be captured by the camera in frames per second.
+  final int? fps;
+
+  /// The video encoding bit rate for recording.
+  final int? videoBitrate;
+
+  /// The audio encoding bit rate for recording.
+  final int? audioBitrate;
```

Comment:
Presumably there are constraints on these values (e.g., non-negative) that should be commented and asserted?

Reply:
commented and asserted.